staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -57,20 +57,7 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 		// TODO: we either want to remove timeout or document it (if we document, move timeout out of this function and declare it in api_installer)
 		timeout := parseTimeout(req.URL.Query().Get("timeout"))
 
-		var (
-			namespace, name string
-			err             error
-		)
-		if includeName {
-			namespace, name, err = scope.Namer.Name(req)
```

Comment:
revert this bit as well... if `includeName` is specified, we must get name info from the request

Reply:
If reverting this, we need to keep this https://github.com/kubernetes/kubernetes/blob/61af419d386dd9aa05ddfd6e192981ae4ab0ffc5/plugin/pkg/admission/noderestriction/admission.go#L364-L367 in noderestriction plugin.

Reply:
hmm... the main thing I care about is keeping the `includeName=true` condition tightly connected to requiring scope.Namer.Name to succeed

what about this:
```
namespace, name, err := scope.Namer.Name(req)
if err != nil {
  if includeName {
    // name was required, return
    scope.err(err, w, req)
    return
  }

  // otherwise attempt to look up the namespace
  namespace, err = scope.Namer.Namespace(req)
  if err != nil {
      scope.err(err, w, req)
      return
  }
}
```

Reply:
if that is satisfied, looking up the name from the object later if needed can be best effort:

```
if len(name) == 0 {
  _, name, _ = scope.Namer.ObjectName(obj)
}
```

Reply:
> if includeName is specified, we must get name info from the request

https://github.com/kubernetes/kubernetes/blob/bb381d6e0af36b56835d7917831d736490b4e7a5/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go#L129-L132

This check has been added.

Reply:
moving it down below allows requests that expected to get name info from the request to instead get it from the object. https://github.com/kubernetes/kubernetes/pull/53185/files#r313210079 preserves the current requirement for `includeName` to be satisfied by the request